Enable GEE with working independence

ailurophilia opened the pull request:
Allow cluster dependence / GEE using the newly added 'cluster' argument for emuFit. Only working independence between rows of Y is supported.

adw96 commented on R/emuFit.R:
```diff
@@ -400,7 +429,8 @@ and the corresponding gradient function to constraint_grad_fn.")
                                          I_inv=I_inv,
-                                         Dy = just_wald_things$Dy)*(n/(n - 1))
+                                         Dy = just_wald_things$Dy,
+                                         cluster = cluster)*score_adj
```
Can you please confirm that score test results will be unchanged from previous versions? I expect that moving from multiplying Dy by this factor to multiplying the score statistic by this factor shouldn't change anything, but would love to confirm as this seems very important.

ailurophilia replied:
So I initially read the (pre-PR) code the same way -- the syntax is not particularly clear, but as things were, we were multiplying the whole score statistic by (n/(n - 1)). (The close paren following Dy is the end of a call to get_score_stat.) So there should be no difference whatsoever!

adw96 commented:
This looks really good, @ailurophilia . 

Can the multiplicative factor `score_adj` be applied in `get_score_stat()`? At present it appears after running this function in `score_test()` and `emuFit()` 

It looks to me like the adjustment is applied every time that function is called -- wouldn't it be more elegant and easier to debug to just build it into `get_score_stat()`?

ailurophilia commented:
Yes great idea – will implement and add to PR!

ailurophilia commented:
Ok! Computation of the`score_adj` factor has been centralized to `get_score_stat`. Doing this broke a test for `score_test` which compared score statistics returned by `score_test` and score statistics directly computed with `get_score_stat`. Because the `score_adj` adjustment was made in `score_test` at the time I wrote the test, I manually computed it for the score statistics returned by `get_score_stat`. Once I updated `get_score_stat` to return a scaled score statistic, the test failed because we now were multiplying the unscaled score statistic returned by `get_score_stat` by `score_adj` twice (once inside the function and once in the body of the test file). Accordingly, I removed the extraneous multiplication in the test file and now the test (appropriately) passes. ✨
